internal/appsec/emitter/httpsec/http.go

```diff
@@ -55,14 +55,13 @@ func ExecuteSDKBodyOperation(parent dyngo.Operation, args types.SDKBodyOperation
 	return err
 }
 
-// WrapHandler wraps the given HTTP handler with the abstract HTTP operation defined by HandlerOperationArgs and
-// HandlerOperationRes.
-// The onBlock params are used to cleanup the context when needed.
-// It is a specific patch meant for Gin, for which we must abort the
-// context since it uses a queue of handlers and it's the only way to make
-// sure other queued handlers don't get executed.
-// TODO: this patch must be removed/improved when we rework our actions/operations system
-func WrapHandler(handler http.Handler, span ddtrace.Span, pathParams map[string]string, onBlock ...func()) http.Handler {
+// WrapHandlerOpts is WrapHandler with support for flexible options.
+func WrapHandlerOpts(handler http.Handler, span ddtrace.Span, pathParams map[string]string, opts ...WrapHandlerOption) http.Handler {
```

Comment:
I suggest we go simpler with:
```suggestion
func WrapHandler(handler http.Handler, span ddtrace.Span, pathParams map[string]string, opts *Config) http.Handler {
```

mainly because:
1. option closures are great for user-friendliness but this is not required for such internal API 
2. in v2, we agreed to move out of this pattern because:
    1. it is not possible to share them easily between package (ie. embedding isn't possible)
    2. they were at the source of several problems in the past few years, including mem leaks due to bad closure scopes, and perf issues due to the underlying struct allocations and deallocation.